rfcs/0000-top-level-browserslist.md

```diff
@@ -0,0 +1,267 @@
+- Repo: `babel/babel`
+- Start Date: 2020-04-25
+- RFC PR: <!-- leave this empty, to be filled in later -->
+- Related Issues: 
+- Authors: [@nicolo-ribaudo](https://github.com/nicolo-ribaudo) 
+- Champion: [@nicolo-ribaudo](https://github.com/nicolo-ribaudo) 
+- Implementors: [@nicolo-ribaudo](https://github.com/nicolo-ribaudo) 
+
+# Summary
+
+This RFC proposes moving the [Browserslist](https://github.com/browserslist/browserslist) integration from `@babel/preset-env` to `@babel/core`. This means deprecating all the Browserslists-related option from the preset and introducing new top-level configuration options.
+
+# Basic example
+
+This example shows how a simple existing configuration can be rewritten using the new API.
+
+```json5
+// Existing
+{
+  "presets": [
+    ["@babel/preset-env", {
+      "targets": ">1%, not ie 11"
+    }]
+  ]
+}
+```
+
+```json5 
+// Proposal
+{
+  "targets": ">1%, not ie 11",
+  "presets": ["@babel/preset-env"]
+}
+```
+
+# Motivation
+
+`targets` is the option that makes `@babel/preset-env` the most powerful preset we have ever had.
+
+However, there are other plugins and presets that would benefit from knowing what engines the user is targeting. The main motivation for me to write this RFC is to be able to share the same targets between `@babel/preset-env` and the [polyfill plugins]().
+
+Currently, you have to specify the targets twice:
+```json5
+// Existing
+{
+  "presets": [
+    ["@babel/preset-env", {
+      "targets": ">1%, not ie 11"
+    }]
+  ],
+  "plugins": [
+    ["polyfill-es-shims", {
+      "targets": ">1%, not ie 11"
+    }]
+  ]
+}
+```
+
+Moving `targets` to the top-level options can also be useful for:
+- plugins that minify the source code, so that they can decide, for exampe, to transform function expressions in arrow functions;
+- generic transform plugins, that can emit a smaller output if they know that the user is targeting modern engines.
+
+# Detailed design
+
+`@babel/preset-env` has 3 Browserslist-related options:
+- [`targets`](https://babeljs.io/docs/en/babel-preset-env#targets)
+- [`configPath`](https://babeljs.io/docs/en/babel-preset-env#configpath)
+- [`ignoreBrowserslistConfig`](https://babeljs.io/docs/en/babel-preset-env#ignorebrowserslistconfig)
+
+[babel/babel#11434](https://github.com/babel/babel/pull/11434), which will land in Babel 7.10.0, introduces a new `browserslistEnv` option.
+
+## New Babel options
+
+### `targets`
```

Comment:
How should `"targets"` be merged? It seems that `[].concat` (like how we did for `presets`) suffices but we should document about that as it may be out of expectation:

If `babel-preset-esmodule` specifies `targets: "chrome 52, safari 14.1, node 13.2"` and `babel-preset-node` specifies `targets: node >= 8`, the resolved `targets` will be `["chrome 52, safari 14.1, node 13.2", "node >= 8"]`, which is the _union_ of these two `targets`.

Reply:
Another edge case for merging targets is that, what if some plugin specifies the `target` as an array whereas another plugin supplies `target` as an object or a string.

Reply:
`targets` are only allowed in user config files, not in presets: presets need to be `targets` _consumers_.

For this reason, I tried thinking what conflicting targets could be caused from, and the most common use cases would be to have an `env` block that defines another set of targets for modern/legacy browsers.

For this reason, I think that _replacing_ targets is what makes most sense.

Reply:
I added these examples to the RFC body